server/plugins/LFUGens.cpp

```diff
@@ -2379,33 +2379,126 @@ void EnvGen_Ctor(EnvGen* unit) {
     EnvGen_next_k(unit, 1);
 }
 
+static bool EnvGen_initSegment(EnvGen* unit, int& counter, double level, float overrideDur = -1) {
+    // Print("stage %d\n", unit->m_stage);
+    // Print("initSegment\n");
+    // out = unit->m_level;
+    int stageOffset = (unit->m_stage << 2) + kEnvGen_nodeOffset;
+
+    if (stageOffset + 4 > unit->mNumInputs) {
+        // oops.
+        Print("envelope went past end of inputs.\n");
+        ClearUnitOutputs(unit, 1);
+        NodeEnd(&unit->mParent->mNode);
+        return false;
+    }
+
+    float previousEndLevel = unit->m_endLevel;
+    float** envPtr = unit->mInBuf + stageOffset;
+    double endLevel = *envPtr[0] * ZIN0(kEnvGen_levelScale) + ZIN0(kEnvGen_levelBias); // scale levels
+    double dur;
+    if (overrideDur >= 0) {
+        dur = overrideDur;
+    } else {
+        dur = *envPtr[1] * ZIN0(kEnvGen_timeScale);
+    }
```

Comment:
you don't need `dur`; you can just name your function parameter to `dur` and set it here if it's nonnegative

Reply:
Sure! Thanks